### PR TITLE
fix: microsoft clarity identify error handling

### PR DESCRIPTION
**Pull request by sanpj2292**
## PR Description

Adding error handling for microsoft clarity destination

## Linear task (optional)

Resolves INT-2937

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Improved error handling for user identification in Microsoft Clarity integration.
	- Asynchronous processing for the identification method enhances performance and reliability.

- **Bug Fixes**
	- Enhanced logging for identification failures to provide better feedback.

### CODE REVIEW

**Review by shrouti1507 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js [hunk 1]
```diff
@@ -55,14 +55,17 @@ class MicrosoftClarity {
...
+      if (context?.traits) {
+        const { traits } = context;
+        const keys = Object.keys(traits);
+        keys.forEach(key => {
+          window.clarity('set', key, traits[key]);
+        });
+      }
+    }).catch(error => {
+      logger.error('Error in identify call', error);
+    });
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:68]: `window.clarity` for `set` will not return a Promise, hence no error handling is needed

**Comment by koladilip** [comment]
Basically we need to set traits call after the identify call, right?

**Comment by sanpj2292** [comment]
> Basically we need to set traits call after the identify call, right?

The previous code was written that way, we are sticking with that approach

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js [hunk 2]
```diff
@@ -55,14 +55,17 @@ class MicrosoftClarity {
...
-      });
-    }
+    window.clarity('identify', userId, sessionId, customPageId).then(() => {
+      if (context?.traits) {
+        const { traits } = context;
+        const keys = Object.keys(traits);
+        keys.forEach(key => {
+          window.clarity('set', key, traits[key]);
+        });
+      }
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:65]: Don't see any requirement of allwoedTraits or sanitization of a value in Microsoft Clarity at this point in time.

**Pull request merged**